### PR TITLE
Discord shop

### PR DESCRIPTION
**Pull request by FloncDev**
Implement the shop into discord

### CODE REVIEW

File: api/src/classes/company.py [hunk 1]
```diff
@@ -181,12 +185,19 @@ def get_achievements(self) -> AchievementsCompanyPublic:
...
+                "id": ach.id,
+                "name": ach.name,
+                "description": ach.description,
+            }
             achievements_details.append(AchievementsCompanyPublic.AchievementSingle.model_validate(details))
 
         # Getting the Company's first and latest Achievement
         company_achievements = sorted(company_achievements, key=lambda x: x.achieved, reverse=False)
-        first_achievement, latest_achievement = company_achievements[0], company_achievements[-1]
+        first_achievement, latest_achievement = (
```
Comment (iauu) [inline api/src/classes/company.py:197]: Possibly `KeyError` if the company have no achievement?

File: bot/src/cogs/shop.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        "item",
+        str,
+        description="The item you want to buy.",
+        autocomplete=item_autocomplete,
+    )
+    @option("quantity", int, description="How many you want to buy, defaults to 1.")
+    async def buy(self, ctx: Context, item: str, quantity: int = 1) -> None:
+        """Buy an item in the shop."""
+        try:
+            company = await self.client.interface.company.get_company(ctx.author.id)
```
Comment (iauu) [inline bot/src/cogs/shop.py:104]: Doesn't need to get the company, you can just give the string user id as `company` in purchase

File: bot/src/wrapper/_communication.py [hunk 3]
```diff
@@ -4,9 +4,9 @@
 
 import aiohttp
 
-from ._api_schema import (  # Company
+from ._api_schema import (
     AchievementGetOutput,  # Achievement
-    AchievementIdGetOutput,
+    AchievementIdGetOutput,  # Company
```
Comment (iauu) [inline bot/src/wrapper/_communication.py:9]: The comment should be one line lower for `# Company` if it is not the first import

File: bot/src/wrapper/schema.py [hunk 4]
```diff
@@ -86,8 +86,8 @@ def from_dict(cls, src: RawAchievement) -> Self:
             name=src["name"],
             description=src["description"],
             companies_earned=src.get("companies_earned", -1),
-            first_achieved=AchievementRecord(**src["first_achieved"]) if src["first_achieved"] else None,
-            latest_achieved=AchievementRecord(**src["latest_achieved"]) if src["latest_achieved"] else None,
+            first_achieved=(AchievementRecord(**src["first_achieved"]) if src["first_achieved"] else None),
```
Comment (iauu) [inline bot/src/wrapper/schema.py:89]: Why adding extra parenthesis?

**Review by i-am-unknown-81514525 — APPROVED** [review]
Just some non-functional impacting issue/question except `api/src/classes/company.py` but should be handle by the API team

File: bot/src/cogs/shop.py (hunk 2 above)
Comment (FloncDev) [reply bot/src/cogs/shop.py:104]: I get the company to make sure they have one, and if they don't I can return an error message

**Pull request merged**